client/blocks/upe/upe-deferred-intent-creation/payment-elements.js

```diff
@@ -19,31 +25,96 @@ import { getBlocksConfiguration } from 'wcstripe/blocks/utils';
  *
  * @return {JSX.Element} Rendered Payment elements.
  */
-const PaymentElements = ( { api, ...props } ) => {
+const PaymentElements = ( {
+	api,
+	supportsDeferredIntent,
+	components: { LoadingMask },
+	...props
+} ) => {
+	const [ clientSecret, setClientSecret ] = useState( null );
+	const [ paymentIntentId, setPaymentIntentId ] = useState( null );
+	const [ hasRequestedIntent, setHasRequestedIntent ] = useState( false );
+
+	useEffect( () => {
+		if ( supportsDeferredIntent || hasRequestedIntent ) {
+			return;
+		}
+
+		async function createIntent() {
+			try {
+				const response = await api.createIntent(
+					getBlocksConfiguration()?.orderId,
+					props.paymentMethodId
+				);
+
+				setClientSecret( response.client_secret );
+				setPaymentIntentId( response.id );
+			} catch ( error ) {
+				// TODO: Gracefully handle errors.
+				// https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3830
+				console.log( 'error', error ); // eslint-disable-line no-console
+			}
+		}
+
+		setHasRequestedIntent( true );
+		createIntent();
+	}, [
+		api,
+		hasRequestedIntent,
+		paymentIntentId,
+		props.paymentMethodId,
+		supportsDeferredIntent,
+	] );
+
+	// If a client secret is required, wait until it is available.
+	if ( ! supportsDeferredIntent && ! clientSecret ) {
+		return (
+			<LoadingMask
+				isLoading={ true }
+				showSpinner={ true }
+				screenReaderLabel={ __(
+					'Loading payment method…',
+					'woocommerce-gateway-stripe'
+				) }
+			/>
+		);
+	}
+
 	const stripe = api.getStripe();
 	const amount = Number( getBlocksConfiguration()?.cartTotal );
 	const currency = getBlocksConfiguration()?.currency.toLowerCase();
 	const appearance = initializeUPEAppearance( api, 'true' );
+
+	// Build options object.
 	const options = {
-		mode: amount < 1 ? 'setup' : 'payment',
-		amount,
-		currency,
-		paymentMethodCreation: 'manual',
-		paymentMethodTypes: getPaymentMethodTypes( props.paymentMethodId ),
 		appearance,
+		paymentMethodCreation: 'manual',
+		fonts: getFontRulesFromPage(),
```

Comment:
Here we're also fixing a missing `fonts` prop which resulted in a minor style issue:

Before:
<img width="840" alt="no-fonts" src="https://github.com/user-attachments/assets/e47a8e41-9ba4-4ea0-90ed-32366888e13f" />

After:
<img width="840" alt="fonts" src="https://github.com/user-attachments/assets/afed5f2c-3d8d-4c0e-b6b3-f8bdc72fe0e1" />